sympy/polys/polytools.py

```diff
@@ -7938,3 +7938,134 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system_poly(polys, **kwargs):
+
+    if not polys:
+        return [[]]
+
+    factored_eqs = []
+    for poly in polys:
+        if not isinstance(poly, Poly):
+            raise TypeError("All inputs must be Poly instances")
+
+        try:
+            constant, factors = factor_list(poly, **kwargs)
+            if not factors and constant.is_zero:
+                return []
+            elif not factors:
+                continue
+            current_factors = []
+            if constant.is_number and constant != 0:
+                current_factors.extend((Poly(factor, poly.gens), []) for factor, _ in factors)
+            else:
+                current_factors.extend((Poly(factor, poly.gens), [Poly(constant, poly.gens)])
+                                       for factor, _ in factors)
+
+            if current_factors:
+                factored_eqs.append(current_factors)
+
+        except ComputationFailed:
+            if kwargs.get('modulus'):
+                raise NotImplementedError("multivariate polynomials over finite fields")
+            raise
+
+    if not factored_eqs:
+        return [[]]
+
+    return _combine_factor_systems_poly(factored_eqs)
+
+
+def _combine_factor_systems_poly(factored_eqs):
+
+    factor_dict = {}
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            key = factor
+            if key not in factor_dict:
+                factor_dict[key] = []
+            factor_dict[key].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    def is_minimal_valid_combination(combo):
+        combo_set = set(combo)
+        # Must satisfy all equations
+        if not all(any(factor in combo_set for factor in eq) for eq in eq_factors):
+            return False
+
+        for size in range(1, len(combo)):
+            for subset in combinations(combo, size):
+                if all(any(factor in set(subset) for factor in eq) for eq in eq_factors):
+                    return False
+        return True
+
+    result = []
+    all_factors = list(factor_dict.keys())
+
+    for size in range(1, len(all_factors) + 1):
+        for combo in combinations(all_factors, size):
+            if is_minimal_valid_combination(combo):
+                combined = [(factor, factor_dict[factor]) for factor in combo]
+                result.append(combined)
+
+    return result
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+    try:
+        if not eqs:
+            return []
+
+        if gens is None:
+            try:
+                gens = list(set().union(*[sympify(eq).free_symbols for eq in eqs]))
+            except (TypeError, AttributeError):
```

Comment:
Don't catch TypeError or AttributeError.

These exceptions should never be caught. Something else in the code should be changed instead.